Common/gui/guibutton.cpp

```diff
@@ -58,7 +58,7 @@ GUIButton::GUIButton()
     Image = -1;
     MouseOverImage = -1;
     PushedImage = -1;
-    CurrentImage = -1;
+    _currentImage = -1;
```

Comment:
Please make _currentImage and flipped private/protected to enforce that they cannot be set without each other, but only using SetCurrentImage method.